src/assets/wise5/components/audioOscillator/audio-oscillator-student/audio-oscillator-student.component.ts

```diff
@@ -215,11 +146,18 @@ class AudioOscillatorController extends ComponentController {
       maxFrequencyPlayed: this.maxFrequencyPlayed,
       submitCounter: this.submitCounter
     };
-    this.createComponentStateAdditionalProcessing(deferred, componentState, action);
-    return deferred.promise;
+
+    const promise = new Promise((resolve, reject) => {
+      this.createComponentStateAdditionalProcessing(
+        { resolve: resolve, reject: reject },
+        componentState,
+        action
+      );
+    });
+    return promise;
```

Comment:
I don't think there's a need to create a new variable here.
```suggestion
    return new Promise((resolve, reject) => {
      this.createComponentStateAdditionalProcessing(
        { resolve: resolve, reject: reject },
        componentState,
        action
      );
    });
```